### PR TITLE
Auto nav code in Protobot

### LINKED ISSUE #3: Copy in Autonomous
https://github.com/Team293/ProtoBot/blob/9c517fa957197b7ba456bf0d06bb59e0d3b29b71/src/main/java/frc/robot/commands/AutonomousCommand.java#L1

We are missing the autonomous code from the previous season. It should be copied in.

### PR DESCRIPTION
**Pull request by karthikappana35**
placed all of the Octabot code in to Protobot.

### CODE REVIEW

**Review by lillaantal1 — APPROVED** [review]

**Review by mahisvivemulapalli — APPROVED** [review]
looks good

**Review by NinaPlazonic — APPROVED** [review]

File: src/main/java/frc/robot/classes/SmoothControl.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    double thetaT = targetPose.getHeadingRadians() - r_angle;
+    thetaT = limitRadians(thetaT); // bound this between -PI to PI
+
+    // Compute the angle between current robot heading and the vector from
+    // the robot to the target
+    double del_r = poseHeading - r_angle;
+    del_r = limitRadians(del_r); // bound this between -PI to PI
+    
+    //Calculate k (equation 14)
+    double k = k(range, thetaT, del_r);
```
Comment (bjmcternan) [inline src/main/java/frc/robot/classes/SmoothControl.java:38]: I'm not a fan of the variable name and function being the same. I did not realize this at the time. We can fix it later

**Review by bjmcternan — APPROVED** [review]
Some very minor cleanup things that we can get another time. This looks good to me

**Review by bjmcternan — CHANGES_REQUESTED** [review]
What were those changes?

**Comment by BobTheLegend** [comment]
> What were those changes?

The autonav code had previously been untested so when testing it I found 2 null pointers as the kinematics and NavX had not been instantiated. In addition, there was a small change in the constants file where a rogue bracket screwed some things up.

**Review by bjmcternan — APPROVED** [review]

**Review by PranjalMhetre — APPROVED** [review]
